modules/coredns/collect.go

```diff
@@ -38,11 +52,20 @@ func (cd *CoreDNS) collect() (map[string]int64, error) {
 
 	mx := newMetrics()
 
-	cd.collectPanic(mx, raw)
-	cd.collectSummaryRequests(mx, raw)
-	//cd.collectSummaryRequestsDuration(mx, raw)
-	cd.collectSummaryRequestsPerType(mx, raw)
-	cd.collectSummaryResponsesPerRcode(mx, raw)
+	// some metric names are different depending on the version
+	// update them once
+	if !cd.skipVersionCheck {
+		cd.updateVersionDependentMetrics(raw)
+		cd.skipVersionCheck = true
+	}
+
+	// we can only get these metrics if we know the server version
+	if cd.version != nil {
+		cd.collectPanic(mx, raw)
+		cd.collectSummaryRequests(mx, raw)
+		cd.collectSummaryRequestsPerType(mx, raw)
+		cd.collectSummaryResponsesPerRcode(mx, raw)
+	}
```

Comment:
Wait, if "cd.version" is nil we can't collect anything because "cd.metricNames" fields are empty strings.